paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -169,41 +205,34 @@ internal abstract class BaseSheetViewModel<TransitionTargetType>(
     @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
     fun setStripeIntent(stripeIntent: StripeIntent?) {
         _stripeIntent.value = stripeIntent
-
```

Comment:
The logic was moved into maps so that the getSupportedMethod was not called more than once.  It will log information if a payment method was removed in the list due to lack of support.  We only want this error message to print once.